src/main/java/com/fauna/response/ConstraintFailure.java

```diff
@@ -1,21 +1,161 @@
 package com.fauna.response;
 
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fauna.constants.ResponseFields;
+import com.fauna.exception.ClientException;
+import com.fauna.exception.ClientResponseException;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 public class ConstraintFailure {
     private final String message;
 
     private final String name;
 
-    private final List<List<Object>> paths;
+    private final PathElement[][] paths;
 
-    public ConstraintFailure(String message, String name, List<List<Object>> paths) {
+    public ConstraintFailure(String message, String name, List<List<Object>> pathLists) {
+        this.message = message;
+        this.name = name;
+        this.paths = new PathElement[pathLists.size()][];
+        for (int i = 0; i < pathLists.size(); i++) {
+            this.paths[i] = new PathElement[pathLists.get(i).size()];
+            for (int j = 0; j < this.paths[i].length; j++) {
+                Object element = pathLists.get(i).get(j);
+                if (element instanceof String) {
+                    paths[i][j] = new PathElement((String) element);
+                } else if (element instanceof Integer) {
+                    paths[i][j] = new PathElement((Integer) element);
+                }
```

Comment:
looks like we were doing the wrong thing in the original test. the integer elements come back as tagged types, which is why we did the pass-through deserializer: `{"@int":"42"}` for example